compiler/llvm/llvmUtil.cpp

```diff
@@ -699,33 +737,45 @@ class LLVMValueTracker : public llvm::AssemblyAnnotationWriter {
 static LLVMValueTracker llvmValueTracker;
 
 void nprint_view(const llvm::Value* arg) {
-  if (arg == NULL) {
-    printf("<NULL>");
-  }
+  printfLLVMHelper("<NULL>");
   else if (const llvm::Function* f = llvm::dyn_cast<llvm::Function>(arg)) {
-    f->print(llvm::outs(), &llvmValueTracker, true, true);
+    auto fd = getLlvmPrintIrFile();
+    if (!fd) fd = &llvm::outs();
```

Comment:
This seems nice. But it might be a good idea in other places too in order to reduce code duplication. Why not have `getLlvmPrintIrFile` return `llvm::outs()` if there isn't a file requested?